src/BSON/Regex.c

```diff
@@ -46,6 +46,38 @@ PHONGO_API zend_class_entry *php_phongo_regex_ce;
 
 zend_object_handlers php_phongo_handler_regex;
 
+/* Initialize the object from a string and return whether it was successful. */
+static bool php_phongo_regex_initialize(php_phongo_regex_t *intern, const char *pattern, phongo_zpp_char_len pattern_len, const char *flags, phongo_zpp_char_len flags_len)
+{
+	intern->pattern = estrndup(pattern, pattern_len);
+	intern->pattern_len = pattern_len;
+	intern->flags = estrndup(flags, flags_len);
+	intern->flags_len = flags_len;
```

Comment:
Again, I think we should stay \0 safe - especially because it's so cheap for us to do.